src/charm.py

```diff
@@ -60,6 +61,7 @@ def __init__(self, *args):
     # logic is moved away from this class.
 
     def on_config_changed(self, event):
+        self.prometheus.render_relation_data()
```

Comment:
Shouldn't you be calling this from the `interface_http.PrometheusInterface.on_relation_joined()`?

Reply:
We should call it both in `on_relation_joined()` and here, since the Prometheus port (and any other config option in the future) might change, so we need to let our remote relations be aware of that. After the databag will be modified by the `render_relation_data`, the `relation-changed` event will be triggered on the remote side by Juju.

Reply:
Ah but then, when Prometheus' configuration changes, Juju actually kills the old unit and creates a new one which, if I'm not mistaken, triggers `interface_http.PrometheusInterface.on_relation_joined()` again. Correct?

Reply:
If that works this way - you're right and we don't have to call it in `on_config_changed`. Let me try it.

Reply:
You were right, it works this way. Removed.